task/attest-api/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! API crate for the root of trust for reporting (RoT-R).
+
+#![no_std]
+
+use derive_idol_err::IdolError;
+use userlib::{sys_send, FromPrimitive};
+
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, IdolError)]
+pub enum AttestError {
```

Comment:
FWIW, should you elect to turn on `encoding: Hubpack`, this type will need to change in the following ways:

1. add `derive(Serialize, Deserialize, hubpack::SerializedSize)`
2. remove `IdolError`, `FromPrimitive`, `#[repr(u32)]`
3. no longer needs explicit discriminator values.